src/kodeagent/tools.py

```
@@ -67,26 +67,64 @@ def calculator(expression: str) -> Union[float, None]:
+                op_type = type(node.op)
+                if op_type not in allowed_operators:
+                    raise ValueError(f'Operator {op_type} not allowed')
+                operand = eval_node(node.operand)
+                return allowed_operators[op_type](operand)
+            else:
+                raise ValueError(f'Unsupported node type: {type(node)}')
+
+        result = eval_node(node)
+        return float(result)
+
+    except Exception as e:
```

Ruff (0.14.6)

Do not catch blind exception: `Exception`

(BLE001)